assets/src/setup/components/user-context-provider.js

```diff
@@ -44,12 +44,17 @@ export function UserContextProvider( { children, userOptionDeveloperTools, userR
 		() => null !== developerToolsOption && developerToolsOption !== originalDeveloperToolsOption,
 		[ developerToolsOption, originalDeveloperToolsOption ],
 	);
-
 	/**
 	 * Fetch user options on mount.
 	 */
 	useEffect( () => {
-		if ( ! originalOptions || false === originalOptions.wizard_completed ) {
+		if ( fetchingOptions ) {
```

Comment:
This sort of does need to run after options are fetched (when `fetchingOptions` is false) because we need to initialize the value based on `wizard_completed`. If `wizard_completed` is false, then we set the dev tools option to null (requiring the user to make a choice) and in this case we don't need to fetch from the user endpoint at all.